### PR TITLE
1541 update node migrate cra to vite

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to ticket

https://leantime.itkdev.dk/#/tickets/showTicket/1569
https://leantime.itkdev.dk/#/tickets/showTicket/1568

#### Description

As it seems [CRA](https://github.com/reactjs/react.dev/pull/5487) is no longer being maintained by fb, I am migrating to [Vite](https://vitejs.dev/).

- upgrade node from 14 -> 20
   - os2display-grid-generator updated as a dependency [was irritating node > 14](https://github.com/os2display/grid-generator/pull/5)
- yarn upgrade
- remove unused dependencies (test-stuff)
- move from CRA/react-app-rewired -> Vite
- Go from [Winston logger](https://github.com/winstonjs/winston) to [Pino logger](https://github.com/pinojs/pino)
  - I tried choose Pino, because it works and because it has a MIT license

#### Screenshot of the result

image

It works!

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

#### Additional comments or questions

- [x] figure out what to do with the [winston-logging](https://github.com/winstonjs/winston), I cannot seem to make it work, and it seems from [Vites github page](https://github.com/vitejs/vite/issues/2618) that they do not intent to make a fix that makes it easy. 

As stated above, Pino will be replacing Winston.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,8 +24,7 @@
     "styled-components": "^5.3.1",
     "suncalc": "^1.9.0",
     "ulid": "^2.3.0",
-    "web-vitals": "^1.0.1",
-    "winston": "^3.3.3"
+    "web-vitals": "^1.0.1"
```
Comment (tuj) [inline package.json]: Can we remove this as well? It is an addition from CRA.

File: src/config-loader.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ const ConfigLoader = {
       ) {
         resolve(configData);
       } else {
-        fetch(`/client/config.json?ts=${nowTimestamp}`)
+        fetch(`config.json?ts=${nowTimestamp}`)
```
Comment (tuj) [inline src/config-loader.js:28]: We have to test how this works on the server...

File: src/release-loader.js [hunk 3]
```diff
@@ -4,7 +4,7 @@
 export default class ReleaseLoader {
   static async loadConfig() {
     const nowTimestamp = new Date().getTime();
-    return fetch(`/client/release.json?ts=${nowTimestamp}`)
```
Comment (tuj) [inline src/release-loader.js:7]: This also requires a test in server setup.

File: src/touch-region.jsx [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-import { ReactComponent as IconClose } from './assets/icon-close.svg';
-import { ReactComponent as IconPointer } from './assets/icon-pointer.svg';
+import { React, useEffect, useState, createRef } from "react";
+import PropTypes from "prop-types";
+import "./touch-region.scss";
+import { createGridArea } from "os2display-grid-generator";
+import Slide from "./slide";
+import ErrorBoundary from "./error-boundary";
+import idFromPath from "./id-from-path";
+import IconClose from "./assets/icon-close.svg?react";
```
Comment (tuj) [inline src/touch-region.jsx:8]: Is this some new way of handling svg components?

**Review by tuj — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (sinejespersen) [reply package.json]: Sure, good catch

File: src/touch-region.jsx (hunk 4 above)
Comment (sinejespersen) [reply src/touch-region.jsx:8]: https://www.npmjs.com/package/vite-plugin-svgr#usage

image

Yes it is vite-plugin-svgr

**Review by tuj — APPROVED** [review]

**Pull request merged**